dallinger/experiment_server/sockets.py

```diff
@@ -23,24 +25,24 @@ class ChatBackend(object):
 
     def __init__(self):
         self.pubsub = conn.pubsub()
+        self._join_pubsub(DEFAULT_CHANNELS)
+        self.clients = defaultdict(list)
+
+    def _join_pubsub(self, channels):
         try:
-            self.pubsub.subscribe(CHANNELS)
-            app.logger.debug('Subscribed to channels: {}'.format(CHANNELS))
+            self.pubsub.subscribe(channels)
+            app.logger.debug(
+                'Subscribed to channels: {}'.format(self.pubsub.channels.keys()))
         except ConnectionError:
             app.logger.exception('Could not connect to redis.')
 
-        self.clients = {}
-        for channel in CHANNELS:
-            self.clients[channel] = []
-
     def subscribe(self, client, channel=None):
         """Register a new client to receive messages."""
         if channel is not None:
-            if channel not in self.clients:
-                raise ValueError('Unknown channel: {}'.format(channel))
             self.clients[channel].append(client)
+            self._join_pubsub([channel])
```

Comment:
I thought that might be the case. Seems like something we should at least have a test for in case the implementation details in the redis library change.